### PR TITLE
Refactor Commands Building and Execution logic

### LINKED ISSUE #36: Refactoring of command execution logic
**Is your feature request related to a problem? Please describe.**
Right now commands are build and executed separately. Also the same command is build in different places making difficult to modify this process.

**Describe the solution you'd like**
A centralized structure that offer a unique API for building and executing commands with solve this problem. Also it must be extensible enough to allow different implementations of this API so it can be used to help in the problems described in #21 and #12.

### PR DESCRIPTION
**Pull request by cbermudez97**
Closes #36 

## Changes:
- Refactor command building and execution to a separate package
- New package core functions only compile to unix systems right now
- Changed commands building and execution api

## Types of changes

What types of changes does your code introduce?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**

- [x] Yes
- [ ] No

**In case you checked yes, did you write tests?**

- [ ] Yes
- [x] No

### CODE REVIEW

File: internal/pkg/commands/cmd_runner_helper_unix.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package utils
+package commands
```
Comment (AntiD2ta) [inline internal/pkg/commands/cmd_runner_helper_unix.go:1]: No build tags here for !windows?

File: internal/pkg/commands/cmd_runner_unix.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
+package commands
```
Comment (AntiD2ta) [inline internal/pkg/commands/cmd_runner_unix.go:1]: Same build tags questions

File: internal/pkg/commands/cmd_runner_helper_unix.go (hunk 1 above)
Comment (cbermudez97) [reply internal/pkg/commands/cmd_runner_helper_unix.go:1]: The name of the file works like the tags. So I removed the unnecessary tags

File: internal/pkg/commands/cmd_runner_unix.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+	} else {
+		log.Info(`Command "docker ps" built without the "--all" flag.`)
+	}
+	return Command{Cmd: command}
+}
+
+func (cr *UnixCMDRunner) BuildDockerComposePSCMD(options DockerComposePsOptions) Command {
+	servs := ""
+	if options.Services {
+		log.Info(`Command "docker-compose ps" built with "--service" flag.`)
```
Comment (AntiD2ta) [inline internal/pkg/commands/cmd_runner_unix.go:39]: Same Debug topic. The thing is that the full command is displayed outside with a log.Info, so I see little point in being redundant in these cases with an Info log too. Is good to log this, but what do you think about changing this to Debug?

File: internal/pkg/commands/cmd_runner_unix.go [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+	command := fmt.Sprintf("docker-compose -f %s up -d %s", options.Path, servs)
+	return Command{Cmd: command}
+}
+
+func (cr *UnixCMDRunner) BuildDockerPSCMD(options DockerPSOptions) Command {
+	command := "docker ps"
+	if options.All {
+		command += " -a"
+	} else {
+		log.Info(`Command "docker ps" built without the "--all" flag.`)
```
[the remark this answers is not in the document]
Comment (cbermudez97) [reply internal/pkg/commands/cmd_runner_unix.go:31]: Sure, on it.

**Review by AntiD2ta — APPROVED** [review]

**Pull request merged**